### ISSUE TITLE
Latest LocalStack fails to invoke a simple AWS Lambda function due to an XML parsing error

### ISSUE THREAD

**Issue by artamonovkirill** [localstack/localstack#3037]
# Type of request: This is a ...

[x] bug report
[ ] feature request

# Detailed description

Lambda invocation has an "unhandled" error with the latest LocalStack, the response payload contains an XML parsing error.

## Expected behavior

No function error is present, the response payload contains no error.

## Actual behavior

Function error is "unhandled". 
The response payload contains the following error and stacktrace:
```
{
	"errorMessage": "Unable to parse response (not well-formed (invalid token): line 1, column 0), invalid XML received. Further retries may succeed:\nb'{}'",
	"errorType": "ResponseParserError",
	"stackTrace": [
	  "  File \"/opt/code/localstack/localstack/services/awslambda/lambda_api.py\", line 541, in run_lambda\n    result = LAMBDA_EXECUTOR.execute(func_arn, func_details, event, context=context,\n", 
	  "  File \"/opt/code/localstack/localstack/services/awslambda/lambda_executors.py\", line 169, in execute\n    return do_execute()\n", 
	  "  File \"/opt/code/localstack/localstack/services/awslambda/lambda_executors.py\", line 161, in do_execute\n    return _run(func_arn=func_arn)\n", 
	  "  File \"/opt/code/localstack/localstack/utils/cloudwatch/cloudwatch_util.py\", line 145, in wrapped\n    publish_error(ns, time_before, e, kwargs)\n", 
	  "  File \"/opt/code/localstack/localstack/utils/cloudwatch/cloudwatch_util.py\", line 130, in publish_error\n    publish_lambda_error(time_before, kwargs)\n", 
	  "  File \"/opt/code/localstack/localstack/utils/cloudwatch/cloudwatch_util.py\", line 46, in publish_lambda_error\n    publish_lambda_metric('Invocations', 1, kwargs)\n", 
	  "  File \"/opt/code/localstack/localstack/utils/cloudwatch/cloudwatch_util.py\", line 30, in publish_lambda_metric\n    cw_client.put_metric_data(Namespace='AWS/Lambda',\n", 
	  "  File \"/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py\", line 337, in _api_call\n    return self._make_api_call(operation_name, kwargs)\n", 
	  "  File \"/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py\", line 642, in _make_api_call\n    http, parsed_response = self._make_request(\n", 
	  "  File \"/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py\", line 662, in _make_request\n    return self._endpoint.make_request(operation_model, request_dict)\n", 
	  "  File \"/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py\", line 102, in make_request\n    return self._send_request(request_dict, operation_model)\n", 
	  "  File \"/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py\", line 134, in _send_request\n    success_response, exception = self._get_response(\n", 
	  "  File \"/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py\", line 166, in _get_response\n    success_response, exception = self._do_get_response(\n", 
	  "  File \"/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py\", line 217, in _do_get_response\n    parsed_response = parser.parse(\n", 
	  "  File \"/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/parsers.py\", line 243, in parse\n    parsed = self._do_error_parse(response, shape)\n", 
	  "  File \"/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/parsers.py\", line 505, in _do_error_parse\n    root = self._parse_xml_string_to_dom(xml_contents)\n", 
	  "  File \"/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/parsers.py\", line 454, in _parse_xml_string_to_dom\n    raise ResponseParserError(\n"]
}
```

# Steps to reproduce

## Command used to start LocalStack

LocalStack is started via TestContainers using [this configuration](https://github.com/artamonovkirill/terraform-localstack/blob/master/src/test/groovy/LocalStack.groovy#L14-L22).

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

Lambda invocation:
* via AWS CLI: `aws lambda invoke help --function-name=test --endpoint http://localhost:38220 response.json`
* via AWS Java SDK: [here](https://github.com/artamonovkirill/terraform-localstack/blob/master/src/test/groovy/LambdaSpec.groovy#L65).

[Lambda code](https://github.com/artamonovkirill/terraform-localstack/blob/master/index.js), [AWS lambda and IAM Terraform config](https://github.com/artamonovkirill/terraform-localstack/blob/master/lambda/main.tf). 

[An example of a test result](https://scans.gradle.com/s/lqhbjf22temd6).

P.S. Same setup works fine with LocalStack 0.11.5.

**Comment by adamkwolf** [comment]
I'm also experiencing this issue

**Comment by kylethorpe** [comment]
Also experiencing this issue, downgraded to 0.11.5 for now.

**Comment by whummer** [comment]
Thanks for reporting @artamonovkirill @adamkwolf @kylethorpe . Can you please add DEBUG=1 to the LocalStack container environment, and share the container logs here as well? Also, is there a way to easily confirm whether the CloudWatch service is available while the test is running? Thanks

**Comment by artamonovkirill** [comment]
@whummer ,
Here are the logs with DEBUG=1:
```
Waiting for all LocalStack services to be ready
2020-10-06 10:56:34,751 CRIT Supervisor is running as root.  Privileges were not dropped because no user is specified in the config file.  If you intend to run as root, you can set user=root in the config file to avoid this message.
2020-10-06 10:56:34,754 INFO supervisord started with pid 14
2020-10-06 10:56:35,725 INFO spawned: 'dashboard' with pid 20
2020-10-06 10:56:35,729 INFO spawned: 'infra' with pid 21
2020-10-06 10:56:35,738 INFO success: dashboard entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2020-10-06 10:56:35,740 INFO exited: dashboard (exit status 0; expected)
(. .venv/bin/activate; exec bin/localstack start --host)
LocalStack version: 0.11.6
Starting local dev environment. CTRL-C to quit.
2020-10-06T10:56:36:DEBUG:bootstrap.py: Loading plugins - scope "services", module "localstack": <function register_localstack_plugins at 0x7fdbdb35fc10>
2020-10-06 10:56:37,524 INFO success: infra entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2020-10-06T10:56:39:INFO:localstack.utils.analytics.profiler: Execution of "load_plugin_from_path" took 2972.3098278045654ms
2020-10-06T10:56:39:INFO:localstack.utils.analytics.profiler: Execution of "load_plugins" took 2972.5470542907715ms
Waiting for all LocalStack services to be ready
Starting edge router (https port 4566)...
Starting mock IAM service on http port 4566 ...
2020-10-06T10:56:41:INFO:localstack.utils.analytics.profiler: Execution of "prepare_environment" took 2045.8459854125977ms
2020-10-06T10:56:41:INFO:localstack.multiserver: Starting multi API server process on port 35529
Running on 0.0.0.0:4566 over https (CTRL + C to quit)
[2020-10-06 10:56:41 +0000] [22] [INFO] Running on 0.0.0.0:4566 over https (CTRL + C to quit)
Running on 0.0.0.0:35529 over http (CTRL + C to quit)
[2020-10-06 10:56:41 +0000] [22] [INFO] Running on 0.0.0.0:35529 over http (CTRL + C to quit)
Starting mock Lambda service on http port 4566 ...
Starting mock CloudWatch Logs service on http port 4566 ...
Starting mock S3 service on http port 4566 ...
Waiting for all LocalStack services to be ready
Ready.
```
I've also put it on [a branch](https://github.com/artamonovkirill/terraform-localstack/tree/debug/localstack) of the project I've mentioned before.

My initial setup there did not include Cloudformation service, but since I've initially faced the problem on a different project with CloudWatch logs enabled, I presume the issue might not be related with CloudWatch not enabled.
Anyhow, I'll modify the setup for my test a bit later today to enable CloudWatch.

**Comment by artamonovkirill** [comment]
@whummer , once I've added "cloudwatch" to the list of services, my test went green: https://github.com/artamonovkirill/terraform-localstack/tree/fix/enable-cloudwatch
Nothing else was changed (e.g., I didn't create the Cloud Watch log group).

Would it be possible to enable "cloudwatch" service if lambda is enabled?
Or somehow stop the container, notifying the user she needs to enable cloudwatch?
Or at least mention the requirement in the documentation and update `serverless` definition, as well as other examples?

Do you want me to create another ticket or update the name and description of this one to pinpoint the root cause?

**Comment by whummer** [comment]
Thanks for confirming the workaround @artamonovkirill @adamkwolf @kylethorpe . This should now be fixed in #3079 - please report here if the problem persists with the `latest` Docker image. Thanks

**Comment by artamonovkirill** [comment]
@whummer thanks for the fix, it works with the `latest` image.
